### PR TITLE
Channels-783 Allow merge tags in TapAction field of mobile push

### PR DESCRIPTION
**Pull request by Sneha-Uppalapati**
[Channels-783](https://segment.atlassian.net/browse/CHANNELS-783) - This PR performs the following:

- allows merge tags in tapAction field of mobile push
- Renames `region` to `settings_region` to avoid double counting

image

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/engage-messaging-twilio/sendMobilePush/push-sender.ts [hunk 1]
```diff
@@ -194,11 +194,25 @@ export class PushSender<Payload extends PushPayload> extends MessageSender<Paylo
...
       },
       profile
     )
 
+    const tapActionButtons = this.payload.customizations?.tapActionButtons ?? []
+    const parsedTapActionButtons = await Promise.all(
+      tapActionButtons.map(async (button) => {
+        return await this.parseContent(
+          {
+            ...button,
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage-messaging-twilio/sendMobilePush/push-sender.ts:208]: We dont have to parse the rest of the fields in button, just the link, so we can be more performant by doing something like

```
return {
  ...button,
  ...(await this.parseContent({ link: button.link })
}
```

**Review by alfrimpong — COMMENTED** [review]
Looks good so far, just 1 comment and unit tests

File: packages/destination-actions/src/destinations/engage-messaging-twilio/__tests__/send-mobile-push.test.ts [hunk 2]
```diff
@@ -570,4 +570,104 @@ describe('sendMobilePush action', () => {
       expect(responses[1].data).toMatchObject(externalIds[0])
     })
   })
+
+  describe('merge tags', () => {
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage-messaging-twilio/__tests__/send-mobile-push.test.ts:574]: Nice! Nit: Can you move these tests to the `liquid template parsing describe` block?

File: packages/destination-actions/src/destinations/engage-messaging-twilio/sendMobilePush/push-sender.ts [hunk 3]
```diff
@@ -209,8 +220,8 @@ export class PushSender<Payload extends PushPayload> extends MessageSender<Paylo
         badgeAmount,
         badgeStrategy,
         media: parsedTemplateContent.media?.length ? parsedTemplateContent.media : undefined,
-        link: this.payload.customizations?.link,
-        tapActionButtons: this.payload.customizations?.tapActionButtons
+        link: parsedTemplateContent.link?.length ? parsedTemplateContent.link : undefined,
+        tapActionButtons: parsedTapActionButtons
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage-messaging-twilio/sendMobilePush/push-sender.ts:224]: if tap action buttons is an empty array, lets set it to null, so the entire field gets dropped from the payload, which would help us stay under 4KB

**Comment by Sneha-Uppalapati** [comment]
> Hi @Sneha-Uppalapati @alfrimpong, there's [another PR](https://github.com/segmentio/action-destinations/pull/1414) currently under review that might potentially result in merge conflicts.

Yes, this will likely be copied into [PR](https://github.com/segmentio/action-destinations/pull/1414) or be merged after his goes in.

**Comment by pmunin** [comment]
@Sneha-Uppalapati would this PR be able to wait until my refactoring PR is merged? https://github.com/segmentio/action-destinations/pull/1414

**Comment by Sneha-Uppalapati** [comment]
> @Sneha-Uppalapati would this PR be able to wait until my refactoring PR is merged? #1414

@pmunin Yes sure, will merge it in after yours.

**Review by marinhero — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @marinhero - can you confirm that I'm OK to merge and deploy this change please? 
I'm not reviewing it.

**Review by marinhero — APPROVED** [review]

**Pull request merged**